### PR TITLE
Add data sources for every type of public DNS record

### PR DESCRIPTION
**Pull request by skatsaounis**
Add public DNS record data sources:

* A record
* AAAA record
* CAA record
* CNAME record
* MX record
* NS record
* PTR record
* SOA record
* SRV record
* TXT record

Add documentation of new data sources and apply minor fixes to documentation of existing DNS resources.

Closes: #14513

### CODE REVIEW

**Comment by skatsaounis** [comment]
@tombuildsstuff Hi! Since you were the one to put the label good-first-issue, could you please take a look at this PR? Thank you in advance 😃

**Comment by skatsaounis** [comment]
@katbyte hi and thank you for pointing out the failures. I fixed them all with my new commit. Feel free to take a look.

```
go test . -v -run "TestAccDataSourceDns*"
=== RUN   TestAccDataSourceDnsARecord_basic
=== PAUSE TestAccDataSourceDnsARecord_basic
=== RUN   TestAccDataSourceDnsAAAARecord_basic
=== PAUSE TestAccDataSourceDnsAAAARecord_basic
=== RUN   TestAccDataSourceDnsCaaRecord_basic
=== PAUSE TestAccDataSourceDnsCaaRecord_basic
=== RUN   TestAccDataSourceDnsCNameRecord_basic
=== PAUSE TestAccDataSourceDnsCNameRecord_basic
=== RUN   TestAccDataSourceDnsMxRecord_basic
=== PAUSE TestAccDataSourceDnsMxRecord_basic
=== RUN   TestAccDataSourceDnsNsRecord_basic
=== PAUSE TestAccDataSourceDnsNsRecord_basic
=== RUN   TestAccDataSourceDnsPtrRecord_basic
=== PAUSE TestAccDataSourceDnsPtrRecord_basic
=== RUN   TestAccDataSourceDnsSoaRecord_basic
=== PAUSE TestAccDataSourceDnsSoaRecord_basic
=== RUN   TestAccDataSourceDnsSrvRecord_basic
=== PAUSE TestAccDataSourceDnsSrvRecord_basic
=== RUN   TestAccDataSourceDnsTxtRecord_basic
=== PAUSE TestAccDataSourceDnsTxtRecord_basic
=== CONT  TestAccDataSourceDnsARecord_basic
=== CONT  TestAccDataSourceDnsNsRecord_basic
=== CONT  TestAccDataSourceDnsCNameRecord_basic
=== CONT  TestAccDataSourceDnsSrvRecord_basic
=== CONT  TestAccDataSourceDnsCaaRecord_basic
=== CONT  TestAccDataSourceDnsSoaRecord_basic
=== CONT  TestAccDataSourceDnsAAAARecord_basic
=== CONT  TestAccDataSourceDnsTxtRecord_basic
--- PASS: TestAccDataSourceDnsSoaRecord_basic (262.10s)
=== CONT  TestAccDataSourceDnsPtrRecord_basic
--- PASS: TestAccDataSourceDnsCaaRecord_basic (262.47s)
=== CONT  TestAccDataSourceDnsMxRecord_basic
--- PASS: TestAccDataSourceDnsAAAARecord_basic (262.73s)
--- PASS: TestAccDataSourceDnsSrvRecord_basic (263.73s)
--- PASS: TestAccDataSourceDnsTxtRecord_basic (272.91s)
--- PASS: TestAccDataSourceDnsPtrRecord_basic (158.11s)
--- PASS: TestAccDataSourceDnsARecord_basic (430.48s)
--- PASS: TestAccDataSourceDnsMxRecord_basic (219.29s)
--- PASS: TestAccDataSourceDnsNsRecord_basic (530.14s)
--- PASS: TestAccDataSourceDnsCNameRecord_basic (531.30s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/dns	532.562s
```

File: website/docs/r/dns_zone.html.markdown [hunk 1]
```diff
@@ -23,38 +23,28 @@ resource "azurerm_dns_zone" "example-public" {
...
 * `minimum_ttl` - (Optional) The minimum Time To Live for the SOA record. By convention, it is used to determine the negative caching duration. Defaults to `300`.
-
 * `refresh_time` - (Optional) The refresh time for the SOA record. Defaults to `3600`.
-
 * `retry_time` - (Optional) The retry time for the SOA record. Defaults to `300`.
-
 * `serial_number` - (Optional) The serial number for the SOA record. Defaults to `1`.
-
 * `ttl` - (Optional) The Time To Live of the SOA Record in seconds. Defaults to `3600`.
-
```
Comment (katbyte) [inline website/docs/r/dns_zone.html.markdown:57]: could we revert this change? it is not consistant with new docs

File: website/docs/d/dns_srv_record.html.markdown [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+* `resource_group_name` - Specifies the resource group where the resource exists.
+* `zone_name` - Specifies the DNS Zone where the DNS Zone (parent resource) exists.
+
+## Attributes Reference
+
+* `id` - The DNS SRV Record ID.
+* `fqdn` - The FQDN of the DNS SRV Record.
+* `ttl` - The Time To Live (TTL) of the DNS record in seconds.
+* `record` - A list of values that make up the SRV record. Each `record` block supports fields documented below.
+* `tags` - A mapping of tags assigned to the resource.
```
Comment (katbyte) [inline website/docs/d/dns_srv_record.html.markdown:41]: could we format all the new docs to be consistent with the rest ?

Comment (katbyte) [reply website/docs/d/dns_srv_record.html.markdown:41]:
```suggestion
* `name` - The name of the DNS SRV Record.

* `resource_group_name` - Specifies the resource group where the resource exists.

* `zone_name` - Specifies the DNS Zone where the DNS Zone (parent resource) exists.

## Attributes Reference

* `id` - The DNS SRV Record ID.

* `fqdn` - The FQDN of the DNS SRV Record.

* `ttl` - The Time To Live (TTL) of the DNS record in seconds.

* `record` - A list of values that make up the SRV record. Each `record` block supports fields documented below.

* `tags` - A mapping of tags assigned to the resource.
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @skatsaounis - just some comments on the docs and then this should be good to merge

**Comment by skatsaounis** [comment]
Hi @katbyte and great thanks for reviewing this beast! 🙇‍♂️ 

I proceeded with adding new lines to every list. That counts for sections: `Attributes Reference`, `Attributes` and `Timeouts`. I did this for every doc file.

The initial code had new lines only to section `Attributes Reference`. If you think I should add another commit to do exactly this instead of what I did please let me know and I will update accordingly!

**Review by katbyte — APPROVED** [review]
Thanks @skatsaounis ! LGTM now 🦕

**Pull request merged**